checks.d/consul.py

```diff
@@ -323,3 +361,69 @@ def check(self, instance):
                         status_value,
                         tags=main_tags+node_tags
                     )
+
+        if perform_network_latency_checks:
+            self.check_network_latency(instance, agent_dc, main_tags)
+
+    def _get_coord_datacenters(self, instance):
+        return self.consul_request(instance, '/v1/coordinate/datacenters')
+
+    def _get_coord_nodes(self, instance):
+        return self.consul_request(instance, 'v1/coordinate/nodes')
+
+    def check_network_latency(self, instance, agent_dc, main_tags):
+
+        datacenters = self._get_coord_datacenters(instance)
+        for datacenter in datacenters:
+            name = datacenter['Datacenter']
+            if name != agent_dc:
+                # Not us, skip
+                continue
+            # Inter-datacenter
+            for other in datacenters:
```

Comment:
Sorry, too many for loops there. I think I see what you're saying in the first one. I'll rework this in the morning.